core/src/nucleus/validation/link_entry.rs

```diff
@@ -33,19 +33,14 @@ pub async fn validate_link_entry(
         }
     };
     let link = link.link().clone();
-    let (base, target) = links_utils::get_link_entries(&link, context).map_err(|_| {
+    let (_base, _target) = links_utils::get_link_entries(&link, context).map_err(|_| {
```

Comment:
So, we don't need base and target anymore to find out what type of link this is, since we have explicit link types now.

We also don't put base and target into the `ValidationData` struct when validating links, so the validation code would (currently) need to explicitly get it anyway if the link validation is dependent of something in the base.

Question: should we either:
* add base and target to `LinkValidationData`
* or remove it completely and also don't treat it as a dependency, which means we could write DNAs that validate a link without having the base
?
@zippy, @artbrock, @maackle

Reply:
I would suggest we keep the hardcoded dependency (which will sub-consciously trigger the pending validation logic) to base and target but add both entries to `LinkValidationData` so that they are available in the validation code, which means DNA devs don't have to `hdk::get_entry` in there to get the linked entries.

Reply:
If we don't want to do that, we could also make the pending validation logic more accessible to DNA devs and have them explicitly say that they depend on base/target, by adding some hdk sugar that returns `ValidationError::UnresolvedDependencies` from the validation callback. (i.e. implementing what we've discussed months ago, @maackle, @artbrock).

Still, I would say we keep hadcoded base/target dependency for links but add them to the validation data now, and maybe take the next step later.

Reply:
Ok, nevermind. We do need base and target for the sub-conscious check of the link valiation anyway. Added that now. I would still suggest to add base and target to `LinkValidationData` - but in another PR.